src/main/java/org/hypertrace/gradle/code/style/CodeStylePlugin.java

```diff
@@ -36,26 +40,34 @@ private void configureFormatting(SpotlessExtension spotlessExtension) {
         });
 
     spotlessExtension.kotlinGradle(
-        format ->
-        {
-            try {
-                format
-                    .ktlint("0.50.0")
-                    .editorConfigOverride(
-                        new HashMap<String, Object>() {
-                          {
-                            put("indent_size", "2");
-                          }
-                        });
-            } catch (IOException e) {
-                throw new RuntimeException(e);
-            }
+        format -> {
+          try {
+            format.ktlint("0.50.0").editorConfigOverride(Map.of("indent_size", "2"));
+          } catch (IOException e) {
+            throw new RuntimeException(e);
+          }
         });
+
+    BufExtension bufExtension = project.getExtensions().getByType(BufExtension.class);
+    spotlessExtension.protobuf(
+        format -> {
+          File bufBinary =
+              project
+                  .getConfigurations()
+                  .getByName(BufSupportKt.BUF_BINARY_CONFIGURATION_NAME)
+                  .getSingleFile();
```

Comment:
We don't typically want to do something like this since it forces the buf binary to be resolved whether the user is formatting or not. If the buf plugin is responsible for downloading it, are they not already making it executable? Can you use a provider instead?

Reply:
Updated to use a provider, let me know if that was what you were expecting.